golem/resources/allocation/allocation.py

```diff
@@ -89,13 +89,10 @@ async def create(cls, node: "GolemNode", data: models.Allocation) -> "Allocation
         return cls(node, created.allocation_id, created)
 
     @api_call_wrapper()
-    async def get_properties_and_constraints_for_demand(
-        self, parser: PayloadSyntaxParser
-    ) -> Tuple[Properties, Constraints]:
+    async def get_properties_and_constraints_for_demand(self) -> Tuple[Properties, Constraints]:
         data = await self.api.get_demand_decorations([self.id])
-
         properties = Properties({prop.key: prop.value for prop in data.properties})
-
-        constraints = Constraints([parser.parse_constraints(c) for c in data.constraints])
-
+        constraints = Constraints(
+            [PayloadSyntaxParser().parse_constraints(c) for c in data.constraints]
```

Comment:
Creating new instance each time, will take significant amount of time, as textx needs to parse `.tx` for its internal state.

Reply:
I doubt it's "significant" in this particular context but okay, moving the instantiation to the constructor above...